### DISCUSSION TITLE
文档协作流程

### DISCUSSION THREAD

**Discussion by juestchaos** [PKM-er/Pkmer-Docs#10] [General]
# 文档协作流程

## 前置条件

1. 本地安装 obsidian 笔记软件。
2. 本地安装 `git` 软件：git 是一款多人协作版本控制系统，能够流程化管理多人的协作项目，可以在 [git 官网](https://git-scm.com/downloads) 下载。
3. 注册 `Github` 帐号：[Github 地址](https://github.com/)

## 协作流程概述

我们将 obsidian 库整个托管在 Github 上，这样任何人都能下载并且修改该库的内容，其基本流程如下：

```mermaid
flowchart TB
A[Pkmer Github项目] -->|Fork到个人项目|B[个人 Github项目] --> |Clone到本地|C[本地Git项目] --> |用Obsidian打开|D[Obsidain库] --> |新建分支|E[修改库内容] --> |git上传修改到个人Github|B --> |在Pkmer Github项目提交PR|A
```

请注意，您最好需要了解什么是 `Git` 再进行操作，[Git - Book](https://git-scm.com/book/zh/v2) 是很好的资源，大致翻看一下有个概念就好了。

1. `Fork` 到个人项目：打开 [Pkmer-Docs](https://github.com/PKM-er/Pkmer-Docs) 链接地址，点击右上角 `Fork` 按钮，此操作会在用户的个人 Github 生成和地址一样的项目，这里称之为个人 Github 项目。
2. `Clone` 到本地：在 Github 网站点击头像，接着点击 `Your Repositories` 跳到你个人项目的页面，找到刚刚 `Fork` 的 `Pkmer-Docs` 项目，进入项目。然后点击绿色按钮 `Code`，复制那一串链接。在本地找个目录，右键打开命令行（git bash），输入 `git clone <刚刚复制的链接>`(不要带尖括号)，这会在该目录下载这个个人项目到本地。
3. 用 obsidian 打开：打开 Obsidian，点击左下角打开其它仓库，打开本地仓库，选中个人项目的位置，即可打开这个项目。
4. `新建分支`：这个 Obsidian 库带有一个 git 插件，`ctrl + p` 打开命令窗口，输入 `create new branch`，回车，接着输入新分支名 (一般是对你想要修改内容的简短总结，如 `doc-md-tutorial` 表示一个 md 教程的文档，记得用英文)。
5. 为 Pkmer 文档添砖加瓦
6. `git` 上传到个人 Github：`ctrl + p` 打开命令窗口，输入并选择 `commit all changes with specific message`，接着输入你做出的修改概要 (如：添加了一个 md 教程)；最后，`ctrl + p` 打开命令窗口，输入并选择`QuickAdd:PushNewBranch`，接着输入分支名，即可完成上传。
7. `PR`：Pull Request 的简称，你可以理解为申请合并到 Pkmer Github 主项目。登录到个人 Github 项目的网站，即第一步的做完后的网址，会有一个 `Pull Request` 的绿色提示，点击后按照指示操作即可。
8. 和项目的维护者沟通，最后维护者会将您的贡献合并到主项目，完成整个贡献流程。

**Comment by windily-cloud** [comment]
## 为什么用git协作

这基于以下几个事实:

1. 显著降低维护成本。纯文本协作有助于网站的维护，可以一键更新最新的网站内容。我们是一群开源贡献者，github这样的成型工作流能显著降低维护压力。任何想要贡献的人，都能在任何时候参与贡献，而不是需要我们预先授权。对于不会使用git的人，可以直接发在issue或者disscuss里参与贡献。
2. 开源理念。我们的项目仓库被公开的让全球访问，大概率不会面临项目审核文件被删，项目存储付费，参与人实名认证等诸多问题，我们有足够的自主权。

目前依旧存在两个问题，一个是git工具对绝大多数人来说是很不友好的，二是github访问国内存在问题。

至于对协作者是否友好的问题，我们可以采取或未来优化以下措施:

1. 协作者直接发issue或者disscuss，由维护者手动更新到项目里
2. 通过github action让项目和gitee同步，从而让国内用户访问gitee参与贡献。
3. 前端开发参与贡献的接口，优化贡献流程，其内容存储依旧在github上
4. 拓展非github的渠道，安排人在飞书，notion，知乎，等多渠道发稿，手动更新到github上

国内访问github的问题，除了自己搞梯子，还有一个[wattToolkit](https://steampp.net/) 软件，全端解决访问github的问题，大多数情况是好用的。
